lib/BuildSystem/BuildNode.cpp

```diff
@@ -126,6 +125,15 @@ FileInfo BuildNode::getLinkInfo(basic::FileSystem& fileSystem) const {
   return fileSystem.getLinkInfo(getName());
 }
 
+basic::CommandSignature BuildNode::getSignature() const {
+  basic::CommandSignature sig;
+  sig.combine(type);
+  for (auto producer : getProducers()) {
```

Comment:
Probably worth a comment here explaining why we do this. I also would like `auto* producer` here so the reader can be easily convinced there is no copy.